scripts/component_hash_update/src/component_hash_update/download.py

```diff
@@ -0,0 +1,374 @@
+    "arm": None,
+}
+
+# TODO: downloads not supported
+# gvisor: sha512 checksums
+# helm_archive: PGP signatures
+# krew_archive: different yaml structure (in our download)
+# calico_crds_archive: different yaml structure (in our download)
+
+# TODO:
+# noarch support -> k8s manifests, helm charts
+# different checksum format (needs download role changes)
+# different verification methods (gpg, cosign) ( needs download role changes) (or verify the sig in this script and only use the checksum in the playbook)
+# perf improvements (async)
+
+def download_hash(downloads: {str: {str: Any}}) -> None:
+    # Handle file with multiples hashes, with various formats.
+    # the lambda is expected to produce a dictionary of hashes indexed by arch name
+    download_hash_extract = {
+            "calicoctl_binary": lambda hashes : {
+                line.split('-')[-1] : line.split()[0]
+                for line in hashes.strip().split('\n')
+                if line.count('-') == 2 and line.split('-')[-2] == "linux"
+                },
+            "etcd_binary": lambda hashes : {
+                line.split('-')[-1].removesuffix('.tar.gz') : line.split()[0]
+                for line in hashes.strip().split('\n')
+                if line.split('-')[-2] == "linux"
+                },
+             "nerdctl_archive": lambda hashes : {
+                line.split()[1].removesuffix('.tar.gz').split('-')[3] : line.split()[0]
+                for line in hashes.strip().split('\n')
+                if [x for x in line.split(' ') if x][1].split('-')[2] == "linux"
+                },
+            "runc": lambda hashes : {
+                parts[1].split('.')[1] : parts[0]
+                for parts in (line.split()
+                              for line in hashes.split('\n')[3:9])
+                },
+             "yq": lambda rhashes_bsd : {
+                 pair[0].split('_')[-1] : pair[1]
+                 # pair = (yq_<os>_<arch>, <hash>)
+                 for pair in ((line.split()[1][1:-1], line.split()[3])
+                     for line in rhashes_bsd.splitlines()
+                     if line.startswith("SHA256"))
+                 if pair[0].startswith("yq")
+                     and pair[0].split('_')[1] == "linux"
+                     and not pair[0].endswith(".tar.gz")
+                },
+            }
+
+    checksums_file = Path(subprocess.Popen(['git', 'rev-parse', '--show-toplevel'],
+                                           stdout=subprocess.PIPE).communicate()[0].rstrip().decode('utf-8')
+                     ) / CHECKSUMS_YML
+    logger.info("Opening checksums file %s...", checksums_file)
+    data, yaml = open_yaml(checksums_file)
+    s = requests.Session()
+
+    @cache
+    def _get_hash_by_arch(download: str, version: str) -> {str: str}:
+
+        hash_file = s.get(downloads[download]['url'].format(
+            version = version,
+            os = "linux",
+            ),
+                          allow_redirects=True)
```

Comment:
nit: the indent looks to a bit weird here, maybe it could be something like that:
```suggestion
        hash_file = s.get(downloads[download]['url'].format(
            version = version,
            os = "linux",
            ), allow_redirects=True)
```

Or other variants of it, btw did you use a formatting tool such as black for those python file? It could be a somewhat easy fix in term to just apply black there.

Reply:
I did not. For some reason I had disabled the the auto lsp formatting w black in my editor, can't remember why exactly, maybe some bugs :thinking: